### PR TITLE
fix(use-resize-observer): resolve type conflict

### LINKED ISSUE #101: Typescript type conflict in dependency `resize-observer-polyfill`
**Describe the bug**
There is a type conflict in the dependency library `resize-observer-polyfill` which causes masonic to fail at build with typescript 4.2.3 or above.
Discussion: que-etc/resize-observer-polyfill#83

Since the `resize-observer-polyfill` is not actively maintained, I suggest switching to https://github.com/juggle/resize-observer. Hopefully this is a drop-in replacement for the old library.

**To Reproduce**

1. Create new react app using the typescript template`npx create-react-app my-app --template typescript`
2. Install `masonic`
3. In `App.tsx`, import masonic
4. Run `npm start`

**Expected behavior**
React app should start at localhost:3000.

**Additional context**
Error message:
```
TypeScript error in /create-react-app/node_modules/resize-observer-polyfill/src/index.d.ts(19,18):
Subsequent property declarations must have the same type.  Property 'contentRect' must be of type 'DOMRectReadOnly', but here has type 'DOMRectReadOnly'.  TS2717

    17 |     interface ResizeObserverEntry {
    18 |         readonly target: Element;
  > 19 |         readonly contentRect: DOMRectReadOnly;
       |                  ^
    20 |     }
    21 |
    22 |     interface ResizeObserver {
```

### PR DESCRIPTION
**Pull request by johnsonsu**
fix typescript 4.2.3 conflict by replacing `resize-observer-polyfill` with `@juggle/resize-observer`

Discussion about the type conflict can be found here: https://github.com/que-etc/resize-observer-polyfill/issues/83.\

All existing tests are passing. No new test added.
Closes #101.

### CODE REVIEW

File: src/use-resize-observer.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
+import { ResizeObserver } from "@juggle/resize-observer";
```
Comment (jaredLunde) [inline src/use-resize-observer.ts:1]: Juggle's `ResizeObserver` is a ponyfill so preferably we'd use the one on the `window` object when it's available

File: src/use-resize-observer.ts [hunk 2]
```diff
@@ -1,11 +1,13 @@
+import { ResizeObserver as Polyfill } from "@juggle/resize-observer";
 import rafSchd from "raf-schd";
 import * as React from "react";
-import ResizeObserver from "resize-observer-polyfill";
 import trieMemoize from "trie-memoize";
 import { elementsCache } from "./elements-cache";
 import { useForceUpdate } from "./use-force-update";
 import type { Positioner } from "./use-positioner";
 
+const ResizeObserver = window.ResizeObserver || Polyfill;
```
Comment (jaredLunde) [inline src/use-resize-observer.ts:9]:
```suggestion
typeof window !== 'undefined' && 'ResizeObserver' in window
    ? window.ResizeObserver
    : Polyfill
```

**Review by jaredLunde — COMMENTED** [review]
One last thing for SSR then we're good. Thanks! Sorry this took so long.

**Comment by johnsonsu** [comment]
> One last thing for SSR then we're good. Thanks! Sorry this took so long.

All good! Thank you for this awesome project!

**Comment by johnsonsu** [comment]
I might be mistaken but I think your edit deleted the variable assignment. I've added it back.

**Comment by jaredLunde** [comment]
whoops!

**Comment by johnsonsu** [comment]
Do you have any idea how to fix this lint error? Lint is passing locally but not the Github Action.

**Comment by joxxoxo** [comment]
I've tried to run lint locally and it fails with the same errors:
```
docker run --rm -it -v "$PWD:/app" -w '/app' node:16-slim bash -c "apt update -qq && apt install -y curl && curl -f https://get.pnpm.io/v6.16.js | node - add --global pnpm && pnpm install && pnpm lint"
```
PS on main repo it's passing

**Comment by joxxoxo** [comment]
Found out that you need to update test, which still references old package and it'll fix linter
https://github.com/johnsonsu/masonic/blob/6ddd1ac6c6872e2c646513f3eff3bf1590f7d6a7/src/use-resize-observer.test.ts#L1

**Comment by joxxoxo** [comment]
Also need to update:
* https://github.com/johnsonsu/masonic/blob/6ddd1ac6c6872e2c646513f3eff3bf1590f7d6a7/types/use-resize-observer.d.ts#L1
* https://github.com/johnsonsu/masonic/blob/6ddd1ac6c6872e2c646513f3eff3bf1590f7d6a7/README.md?plain=1#L55
* https://github.com/johnsonsu/masonic/blob/6ddd1ac6c6872e2c646513f3eff3bf1590f7d6a7/docs/v2.md?plain=1#L53

**Comment by johnsonsu** [comment]
Thank you for looking into it @joxxoxo. I've updated those files. Pending @jaredLunde to trigger the workflow.

**Comment by joxxoxo** [comment]
@johnsonsu awesome, looking forward to have TS passing on my project 😀
Have you checked tests locally? I think it may need some adjustments to pass

**Comment by jaredLunde** [comment]
Thanks for putting this together! I've been super busy lately and I very much appreciate the community effort.

**Pull request merged**